servicesContracts.py

```diff
@@ -1,10 +1,28 @@
 from typing import Any
-
-from artifacts import *
 from servicesContracts.requestEthereum_service import RequestEthereumService
 
+"""
+  getServiceFromAddress return the service of a coresponding currency contract address
+  :param  address:
+        The address of the currency contract
+"""
 def getServiceFromAddress(address: str) -> Any:
-    pass
+    if not address:
+        return None
+    if isThisArtifact(Artifacts.requestEthereumArtifact, address):
+        return RequestEthereumService()
 
+"""
+  isThisArtifact return
+  :param  artifact:
+      RequestNetwork Artifact to use in its interactions with the Ethereum network
+    :param  address:
+        The address of the currency contract
+"""
 def isThisArtifact(artifact, address: str) -> bool:
-    pass
\ No newline at end of file
+    if not address:
+        return False
+    sanitizedAdress = address.lower()
+    # assume artifact.networks is a dictionary
+    return any((value.address.lower() == sanitizedAdress if not value.address else False
```

Comment:
This line is complicated and should be broken apart so that other people have an easier time reading it. Also the `artifact.networks` won't work, you need `artifact["networks"]`

Reply:
I changed into a regular For loop.